### PR TITLE
2903 - update NR to use legal api proxy endpoint

### PR DESCRIPTION
**Pull request by ochiu**
*Issue #:* /bcgov/entity#2903

*Description of changes:*
- Renamed namex-api-mixin to name-request-mixin
- Removed old temporary token code
- Added query name request call to legal-api-mixin
- Moved NR Authorizations helper into name-request-mixin
- Updated NR expiration status check to look at expirationDate rather than status

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

File: src/mixins/name-request-mixin.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    }
+
+    return { isConsumable: true, expired: false, approved: true }
+  }
+
+  /**
+   * Fetch Authorizations by NR number
+   * @param nr the name request number (e.g NR 1234567)
+   */
+  getNRAuthorizations (nrNumber: string): Promise<any> {
```
Comment (severinbeauvais) [inline src/mixins/name-request-mixin.ts:48]: Might make sense to move this to Legal API mixin. Thoughts?

File: src/App.vue [hunk 2]
```diff
@@ -220,7 +209,6 @@ export default class App extends Mixins(DateMixin, FilingTemplateMixin, LegalApi
 
     return this.getNRAuthorizations(queryNrNumber).then(async data => {
       this.storeAuthRoles(data) // throws if no role
-      await this.intializeNameXToken() // TODO : temporary token retrieval code
       const nrResponse = await this.queryNameRequest(queryNrNumber)
 
       // NR not found
```
Comment (severinbeauvais) [inline src/App.vue:214]: In Business Filings UI, I added some extra validity checks that you might also want to implement.
See https://github.com/bcgov/business-filings-ui/blob/cb421eed27d97bc63dc99596d42277d3be7e0726/src/App.vue#L422

File: src/mixins/name-request-mixin.ts (hunk 1 above)
Comment (ochiu) [reply src/mixins/name-request-mixin.ts]: I was pondering this as well, so good to have a second opinion. I moved it to the legal api mixin now.

File: src/App.vue (hunk 2 above)
Comment (ochiu) [reply src/App.vue:214]: Thanks! I added that function to the name request mixin and added a state to the NR invalid dialog as well as updated the tests.

**Pull request merged**